pkg/generate/app/cmd/newapp_test.go

```diff
@@ -415,6 +415,7 @@ func TestDetectSource(t *testing.T) {
 	dockerResolver := app.DockerRegistryResolver{
 		Client: dockerregistry.NewClient(),
 	}
+	mocks := app.MockSourceRepositories()
```

Comment:
Yes, that mutation is needed here. In the process of detecting the source, `detectSource` [edits](https://github.com/openshift/origin/blob/719543c31713c49fc449553f5571f4c6d32631e4/pkg/generate/app/cmd/newapp.go#L347) (see [UsedBy](https://github.com/openshift/origin/blob/719543c31713c49fc449553f5571f4c6d32631e4/pkg/generate/app/sourcelookup.go#L57-L60)) the source repository and I need that edit to propagate to the mock repository. Unfortunately we cannot mock that since if the repo is [in use](https://github.com/openshift/origin/blob/719543c31713c49fc449553f5571f4c6d32631e4/pkg/generate/app/cmd/newapp.go#L279) in the first place, we wont get a source detection for it.